linera-core/src/unit_tests/test_utils.rs

```diff
@@ -335,69 +335,34 @@ where
         proposal: BlockProposal,
         sender: oneshot::Sender<Result<ChainInfoResponse, NodeError>>,
     ) -> Result<(), Result<ChainInfoResponse, NodeError>> {
-        let mut validator = self.client.lock().await;
-        let handle_block_proposal_result =
-            Self::handle_block_proposal(proposal, &mut validator).await;
-        let result = match handle_block_proposal_result {
-            Some(Err(NodeError::BlobsNotFound(_))) => {
-                handle_block_proposal_result.expect("handle_block_proposal_result should be Some")
-            }
-            _ => match validator.fault_type {
```

Comment:
I don't understand why you couldn't do the same thing w/o moving the `FaultType` to the new type? If we moved the match on `validator.fault_type` to the beginning of this function (and the other one), and only call to `handle_block_proposal` for appropriate "faults" – wouldn't that fix the problem as well?

I am not sure a *client is a good place to store the validator's fault.

Reply:
Since this is already inside the spawned task I think that wouldn't be enough.

Maybe `LocalValidatorClient` isn't a great name anyway: It represents the handle of a connection to the validator, on the client side. But in these tests, since we don't actually have a network, that distinction isn't clear anyway. And I'd say it doesn't matter whether we store the fault type inside that handle or inside `LocalValidator`.